.circleci/config.yml

```diff
@@ -1,22 +1,150 @@
-version: 2
+version: 2.1
+workflows:
+  build_and_deploy:
+    jobs:
+      - build:
+          filters:
+            tags:
+              only: /.*/
+      - test-python-install:
+          version: "3.6"
+          requires:
+            - build
+      - deploy:
+          requires:
+            - build
+          filters:
+            tags:
+              only: /[0-9]+(\.[0-9]+)*/
+            branches:
+              ignore: /.*/
+      - smoke_test:
+          requires:
+            - build
+            - deploy
+          filters:
+            tags:
+              only: /[0-9]+(\.[0-9]+)*/
+            branches:
+              ignore: /.*/
 jobs:
   build:
     docker:
-    - image: circleci/python:3.6
+      - image: circleci/python:3.6
     steps:
-    - checkout
-    - run:
-        name: Setup testing environment
-        command: |
-          make install-dev
-    - run:
-        name: Run tests and measure coverage
-        command: |
-          make coverage
-    - store_artifacts:
-        path: htmlcov
-workflows:
-  version: 2
-  workflow:
-    jobs:
-    - build
+      - checkout
+      - restore_cache:
+          key: v1-dependency-cache-{{ checksum "setup.py" }}-{{ checksum "Makefile" }}
+      - run:
+          name: Install Python Dependencies
+          command: |
+            python3 -m venv venv
```

Comment:
Why are we using a virtual environment in CI?

Reply:
The authors of the python article for CircleCI were recommending doing it.  Since the docker image is being reused during the session, it makes some sense to put some items into a virtual env.  I'm not wed to the idea, but I see merit in doing it.

Reply:
Fair enough